File: 2729-check-if-the-number-is-fascinating/2729-check-if-the-number-is-fascinating.py
class Solution:
  def isFascinating(self, n: int) -> bool:
    num = str(n) + str(n * 2) + str(n * 3)
    hash_map = dict()
    
    for digit in num:
      hash_map[digit] = hash_map.get(digit, 0) + 1

    for i in range(1, 10):
      if hash_map.get(str(i), 0) != 1:
        return False
    
    return True
    
    
        

File: 2729-check-if-the-number-is-fascinating/2729-check-if-the-number-is-fascinating.cpp
class Solution {
public:
  bool isFascinating(int n) {
    unordered_map<char, int> hash_map;
    string num = to_string(n) + to_string(n * 2) + to_string(n * 3);
    
    for (auto& c: num) hash_map[c]++;
    
    for (int i = 1; i < 10; i++) 
      if (hash_map['0' + i] != 1)
        return false;
    
    return true;
  }
};